### DISCUSSION TITLE
Provider of providers??

### DISCUSSION THREAD

**Discussion by najibghadri** [rrousselGit/riverpod#923] [Q&A] [answered]
In a chat app I have a conversation_list_provider that manages the list of conversations, what is the last message for each and showing/creating new conversations. BUT we think that during the lifecycle of the app, each conversation should have its own conversation_provider which manages the whole message list, paging and keeping the messages in memory.

This means that we need a collection of conversation_provider, ie multiple instances of the same StateNotifier class. My question is, is it valid to have a provider that manages these providers of the same type??

OR should I always have only one provider instance of a provider type (a Notifier class) and deal with storage in a different way?

**Comment by rrousselGit** [comment, marked as answer]
You shouldn't dynamically create providers, so no

Instead maybe you want to use `family`

**Reply by najibghadri** [reply]
Thank you very much @rrousselGit ! To me it seems like `family` is exactly what I need bc it's kind of like creating provider dynamically, and I suppose I can create as many as I want.
